pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -382,13 +382,66 @@ func (pt *PipelineTask) GetMatrixCombinationsCount() int {
 	if !pt.IsMatrixed() {
 		return 0
 	}
+	// Explicit combinations in the Matrix
+	// Calculate based on number of include Names and return count
+	if pt.Matrix.MatrixHasInclude() && !pt.Matrix.MatrixHasParams() {
+		var count = 0
+		// Iterate over each include.Name that has params
+		for _, include := range pt.Matrix.Include {
+			for _, includeParams := range include.Params {
+				// Increment count for every include name
+				if includeParams.Name != "" && includeParams.Value.StringVal != "" {
+					count++
+					break
+				}
+			}
+		}
+		return count
+	}
+	// Create a set of param names
+	paramNames := make(map[string][]string)
 	count := 1
+	// Iterate over matrix params and generate all combinations
 	for _, param := range pt.Matrix.Params {
 		count *= len(param.Value.ArrayVal)
+		paramNames[param.Name] = param.Value.ArrayVal
+	}
+	if pt.Matrix.MatrixHasInclude() {
+		// Filter out which include params will be replaced
+		for _, include := range pt.Matrix.Include {
+			for _, param := range include.Params {
+				// Check if specific param name is in mapped paramNames
+				if val, exist := paramNames[param.Name]; exist {
+					// If param value exists in mapping, it will overwrite existing param
+					// No new combinations generated
+					if contains(val, param.Value.StringVal) {
+						break
+					}
+				} else {
+					// Param name not found in mapping
+					// Common param will overwrite all params
+					// No new combinations generated
+					break
+				}
+				// Otherwise the param value does not exist
+				// New combination generated
+				count++
+			}
+		}
 	}
 	return count
 }
 
+// Contains returns true if a string exists in a slice
+func contains(s []string, str string) bool {
```

Comment:
Please try to utilize the newly introduced `slices.Contains()` instead of creating such function: https://golangtutorial.dev/tips/golang-slice-contains-method/